{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fe08097d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd, json, pickle\n",
    "from multiprocessing import Pool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bf021e5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "original_df = pd.read_csv('blood-samples/full-data.csv', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "62d20f29",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>car_chr</th>\n",
       "      <th>car_start</th>\n",
       "      <th>car_end</th>\n",
       "      <th>car_size</th>\n",
       "      <th>car_intlogqvalue</th>\n",
       "      <th>car_foldchange</th>\n",
       "      <th>car_pvalue</th>\n",
       "      <th>car_qvalue</th>\n",
       "      <th>car_summit_to_start</th>\n",
       "      <th>car_sample_id</th>\n",
       "      <th>...</th>\n",
       "      <th>car_se_anno</th>\n",
       "      <th>car_methylation_anno</th>\n",
       "      <th>car_interaction_anno</th>\n",
       "      <th>car_commmon_snp_anno</th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Biosample type</th>\n",
       "      <th>Biosample name</th>\n",
       "      <th>Tissue type</th>\n",
       "      <th>Cancer type</th>\n",
       "      <th>Region number</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>9997</td>\n",
       "      <td>10802</td>\n",
       "      <td>805</td>\n",
       "      <td>1359</td>\n",
       "      <td>13.36316</td>\n",
       "      <td>138.88042</td>\n",
       "      <td>135.96225</td>\n",
       "      <td>204</td>\n",
       "      <td>25</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>Sample_0025</td>\n",
       "      <td>Primary Cell</td>\n",
       "      <td>CD56Bright natural killer cell</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Normal</td>\n",
       "      <td>48025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20963</td>\n",
       "      <td>21340</td>\n",
       "      <td>377</td>\n",
       "      <td>108</td>\n",
       "      <td>5.73316</td>\n",
       "      <td>13.03911</td>\n",
       "      <td>10.84643</td>\n",
       "      <td>329</td>\n",
       "      <td>25</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Sample_0025</td>\n",
       "      <td>Primary Cell</td>\n",
       "      <td>CD56Bright natural killer cell</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Normal</td>\n",
       "      <td>48025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28785</td>\n",
       "      <td>29747</td>\n",
       "      <td>962</td>\n",
       "      <td>1636</td>\n",
       "      <td>18.39825</td>\n",
       "      <td>166.64635</td>\n",
       "      <td>163.63165</td>\n",
       "      <td>551</td>\n",
       "      <td>25</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>189</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>Sample_0025</td>\n",
       "      <td>Primary Cell</td>\n",
       "      <td>CD56Bright natural killer cell</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Normal</td>\n",
       "      <td>48025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>437449</td>\n",
       "      <td>437967</td>\n",
       "      <td>518</td>\n",
       "      <td>106</td>\n",
       "      <td>6.49730</td>\n",
       "      <td>12.88990</td>\n",
       "      <td>10.69930</td>\n",
       "      <td>515</td>\n",
       "      <td>25</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Sample_0025</td>\n",
       "      <td>Primary Cell</td>\n",
       "      <td>CD56Bright natural killer cell</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Normal</td>\n",
       "      <td>48025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>540776</td>\n",
       "      <td>541301</td>\n",
       "      <td>525</td>\n",
       "      <td>263</td>\n",
       "      <td>9.65766</td>\n",
       "      <td>28.77254</td>\n",
       "      <td>26.39556</td>\n",
       "      <td>193</td>\n",
       "      <td>25</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>Sample_0025</td>\n",
       "      <td>Primary Cell</td>\n",
       "      <td>CD56Bright natural killer cell</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Normal</td>\n",
       "      <td>48025</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  car_chr  car_start  car_end  car_size  car_intlogqvalue  car_foldchange  \\\n",
       "0    chr1       9997    10802       805              1359        13.36316   \n",
       "1    chr1      20963    21340       377               108         5.73316   \n",
       "2    chr1      28785    29747       962              1636        18.39825   \n",
       "3    chr1     437449   437967       518               106         6.49730   \n",
       "4    chr1     540776   541301       525               263         9.65766   \n",
       "\n",
       "   car_pvalue  car_qvalue  car_summit_to_start  car_sample_id  ...  \\\n",
       "0   138.88042   135.96225                  204             25  ...   \n",
       "1    13.03911    10.84643                  329             25  ...   \n",
       "2   166.64635   163.63165                  551             25  ...   \n",
       "3    12.88990    10.69930                  515             25  ...   \n",
       "4    28.77254    26.39556                  193             25  ...   \n",
       "\n",
       "   car_se_anno car_methylation_anno  car_interaction_anno  \\\n",
       "0            0                    0                     1   \n",
       "1            0                    0                     0   \n",
       "2            0                  189                     3   \n",
       "3            0                    0                     0   \n",
       "4            2                    0                     6   \n",
       "\n",
       "   car_commmon_snp_anno    Sample ID  Biosample type  \\\n",
       "0                     5  Sample_0025    Primary Cell   \n",
       "1                     0  Sample_0025    Primary Cell   \n",
       "2                     1  Sample_0025    Primary Cell   \n",
       "3                     0  Sample_0025    Primary Cell   \n",
       "4                     4  Sample_0025    Primary Cell   \n",
       "\n",
       "                   Biosample name  Tissue type  Cancer type  Region number  \n",
       "0  CD56Bright natural killer cell        Blood       Normal          48025  \n",
       "1  CD56Bright natural killer cell        Blood       Normal          48025  \n",
       "2  CD56Bright natural killer cell        Blood       Normal          48025  \n",
       "3  CD56Bright natural killer cell        Blood       Normal          48025  \n",
       "4  CD56Bright natural killer cell        Blood       Normal          48025  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "original_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "606cbc09",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['car_chr', 'car_start', 'car_end', 'car_size', 'car_intlogqvalue',\n",
       "       'car_foldchange', 'car_pvalue', 'car_qvalue', 'car_summit_to_start',\n",
       "       'car_sample_id', 'car_region_id', 'car_gsm_id', 'car_risk_snp_anno',\n",
       "       'car_eqtl_anno', 'car_crisps_anno', 'car_te_anno', 'car_se_anno',\n",
       "       'car_methylation_anno', 'car_interaction_anno', 'car_commmon_snp_anno',\n",
       "       'Sample ID', 'Biosample type', 'Biosample name', 'Tissue type',\n",
       "       'Cancer type', 'Region number'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "original_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ad1c4a75",
   "metadata": {},
   "outputs": [],
   "source": [
    "genome = pickle.load(open('hg19.pickle', 'rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e787f919",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['chr1', 'chr2', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', 'chr20', 'chr21', 'chr22', 'chrX', 'chrY'])"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genome.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bf89c863",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n",
      "{'G', 'C', 'T', 'N', 'A'}\n"
     ]
    }
   ],
   "source": [
    "for key, s in genome.items():\n",
    "    print(set(s))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3fdb170d",
   "metadata": {},
   "outputs": [],
   "source": [
    "sample25 = original_df[original_df['Sample ID'] == 'Sample_0025']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ed610594",
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_megaseq(sample_id):\n",
    "    bigseq = []\n",
    "    \n",
    "    sample = original_df[original_df['Sample ID'] == sample_id]\n",
    "\n",
    "    for idx, row in sample.iterrows():\n",
    "        chromosome = row['car_chr']\n",
    "        start = row['car_start']\n",
    "        end = row['car_end']\n",
    "        sequence = genome[chromosome][start:end+1]\n",
    "        bigseq.append(sequence)\n",
    "        #if idx < 10:\n",
    "            #print(start, end)\n",
    "            #print(end-start+1)\n",
    "            #print(sequence)\n",
    "\n",
    "    megaseq = 'Z'.join(bigseq)\n",
    "    \n",
    "    return megaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f932f39b",
   "metadata": {},
   "outputs": [],
   "source": [
    "megaseq25 = generate_megaseq('Sample_0025')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db98ee08",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "be8cb381",
   "metadata": {},
   "outputs": [],
   "source": [
    "samples_frame = pd.read_csv('Samples-Of-Interest.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e5c1e25d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1680fc4a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1, 4, 9]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\"\\n#df_of_megaseqs = pd.Dataframe(columns=['Sample ID', 'sequence'])\\nms_dfs = []\\n\\nfor idx, row in samples_frame.iterrows():\\n    sample_mega = generate_megaseq(row['Sample ID'])\\n    df_here = pd.DataFrame(columns=['Sample ID', 'sequence'])\\n    df_here.loc[1] = [row['Sample ID'], sample_mega]\\n    ms_dfs.append(df_here)\\n    \\ndf_of_megaseqs = pd.concat(ms_dfs)\\n\""
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def f(x):\n",
    "    return x*x\n",
    "\n",
    "with Pool(5) as p:\n",
    "    print(p.map(f, [1, 2, 3]))\n",
    "    \n",
    "sample_names = [row['Sample ID'] for idx, row in samples_frame.iterrows()]\n",
    "    \n",
    "    \n",
    "with Pool(16) as wickedpool:\n",
    "    megaseqs = wickedpool.map(generate_megaseq, sample_names)\n",
    "    \n",
    "def make_df(sample_name, mega):\n",
    "    df_here = pd.DataFrame(columns=['Sample ID', 'sequence'])\n",
    "    df_here.loc[1] = [sample_name, mega]\n",
    "    return df_here\n",
    "\n",
    "inputs_here = zip(sample_names, megaseqs)\n",
    "    \n",
    "with Pool(16) as anotherpool:\n",
    "    dfs = anotherpool.starmap(make_df, inputs_here)\n",
    "    \n",
    "\n",
    "df_of_megaseqs = pd.concat(dfs)\n",
    "\n",
    "'''\n",
    "#df_of_megaseqs = pd.Dataframe(columns=['Sample ID', 'sequence'])\n",
    "ms_dfs = []\n",
    "\n",
    "for idx, row in samples_frame.iterrows():\n",
    "    sample_mega = generate_megaseq(row['Sample ID'])\n",
    "    df_here = pd.DataFrame(columns=['Sample ID', 'sequence'])\n",
    "    df_here.loc[1] = [row['Sample ID'], sample_mega]\n",
    "    ms_dfs.append(df_here)\n",
    "    \n",
    "df_of_megaseqs = pd.concat(ms_dfs)\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "10f521b7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 452 entries, 1 to 1\n",
      "Data columns (total 2 columns):\n",
      " #   Column     Non-Null Count  Dtype \n",
      "---  ------     --------------  ----- \n",
      " 0   Sample ID  452 non-null    object\n",
      " 1   sequence   452 non-null    object\n",
      "dtypes: object(2)\n",
      "memory usage: 10.6+ KB\n"
     ]
    }
   ],
   "source": [
    "df_of_megaseqs.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "075e4f7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "mega25 = df_of_megaseqs[['sequence']].iloc[0]['sequence']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "af0ace68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "33278645"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(mega25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "832a24c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'A', 'C', 'G', 'N', 'T', 'Z'}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(mega25)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "04c0f592",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G count 9303278\t percent    27.9557 %\n",
      "C count 9282473\t percent    27.8932 %\n",
      "T count 7318978\t percent    21.9930 %\n",
      "Z count 48024\t percent     0.1443 %\n",
      "N count 20\t percent     0.0001 %\n",
      "A count 7325872\t percent    22.0137 %\n"
     ]
    }
   ],
   "source": [
    "for item in set(mega25):\n",
    "    print(f'{item} count {mega25.count(item)}\\t percent {\"{:10.4f}\".format(100*mega25.count(item)/len(mega25))} %')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9f06c4fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_of_megaseqs = df_of_megaseqs.set_index('Sample ID')\n",
    "samples_frame = samples_frame.set_index('Sample ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "c1595fb8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Sample_0025</th>\n",
       "      <td>NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0026</th>\n",
       "      <td>NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0027</th>\n",
       "      <td>AACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0028</th>\n",
       "      <td>CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0029</th>\n",
       "      <td>NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1467</th>\n",
       "      <td>CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1468</th>\n",
       "      <td>CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1469</th>\n",
       "      <td>ACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1470</th>\n",
       "      <td>CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1471</th>\n",
       "      <td>CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>452 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      sequence\n",
       "Sample ID                                                     \n",
       "Sample_0025  NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...\n",
       "Sample_0026  NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...\n",
       "Sample_0027  AACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACC...\n",
       "Sample_0028  CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...\n",
       "Sample_0029  NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...\n",
       "...                                                        ...\n",
       "Sample_1467  CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...\n",
       "Sample_1468  CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...\n",
       "Sample_1469  ACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCC...\n",
       "Sample_1470  CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...\n",
       "Sample_1471  CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...\n",
       "\n",
       "[452 rows x 1 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_of_megaseqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "6922ef29",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_of_megaseqs['Cancer type'] = samples_frame['Cancer type']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "fdced4bd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Myeloma\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Cultured cancer\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "The Yellow fever\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "HIV-1\n",
      "HIV-1\n",
      "HIV-1\n",
      "HIV-1\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Melanoma\n",
      "Melanoma\n",
      "Leukemia\n",
      "Leukemia\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Leukemia\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Lymphoma\n",
      "Lymphoma\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Coronary artery disease\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Normal\n",
      "Hypogammaglobulinemia\n",
      "Hypogammaglobulinemia\n",
      "Hypogammaglobulinemia\n",
      "Hypogammaglobulinemia\n",
      "Hypogammaglobulinemia\n",
      "Normal\n",
      "Normal\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n",
      "Other\n"
     ]
    }
   ],
   "source": [
    "for t in df_of_megaseqs['Cancer type']:\n",
    "    print(t)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ebc923b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "      <th>Cancer type</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Sample_0025</th>\n",
       "      <td>NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0026</th>\n",
       "      <td>NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0027</th>\n",
       "      <td>AACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACC...</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0028</th>\n",
       "      <td>CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0029</th>\n",
       "      <td>NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1467</th>\n",
       "      <td>CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1468</th>\n",
       "      <td>CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1469</th>\n",
       "      <td>ACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCC...</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1470</th>\n",
       "      <td>CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1471</th>\n",
       "      <td>CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...</td>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>452 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      sequence Cancer type\n",
       "Sample ID                                                                 \n",
       "Sample_0025  NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...      Normal\n",
       "Sample_0026  NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...      Normal\n",
       "Sample_0027  AACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACC...      Normal\n",
       "Sample_0028  CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...      Normal\n",
       "Sample_0029  NNNTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...      Normal\n",
       "...                                                        ...         ...\n",
       "Sample_1467  CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...       Other\n",
       "Sample_1468  CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...       Other\n",
       "Sample_1469  ACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCC...       Other\n",
       "Sample_1470  CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...       Other\n",
       "Sample_1471  CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT...       Other\n",
       "\n",
       "[452 rows x 2 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_of_megaseqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "0fe456fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Biosample type</th>\n",
       "      <th>Biosample name</th>\n",
       "      <th>Tissue type</th>\n",
       "      <th>Cancer type</th>\n",
       "      <th>Region number</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Sample_0025</th>\n",
       "      <td>Primary Cell</td>\n",
       "      <td>CD56Bright natural killer cell</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Normal</td>\n",
       "      <td>48025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0026</th>\n",
       "      <td>Primary Cell</td>\n",
       "      <td>CD56Bright natural killer cell</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Normal</td>\n",
       "      <td>50447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0027</th>\n",
       "      <td>Primary Cell</td>\n",
       "      <td>CD56Bright natural killer cell</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Normal</td>\n",
       "      <td>47349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0028</th>\n",
       "      <td>Primary Cell</td>\n",
       "      <td>CD56Bright natural killer cell</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Normal</td>\n",
       "      <td>46698</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_0029</th>\n",
       "      <td>Primary Cell</td>\n",
       "      <td>CD56Dim natural killer cell</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Normal</td>\n",
       "      <td>40745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1467</th>\n",
       "      <td>Other</td>\n",
       "      <td>Macrophage</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Other</td>\n",
       "      <td>10568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1468</th>\n",
       "      <td>Other</td>\n",
       "      <td>Monocyte</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Other</td>\n",
       "      <td>6102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1469</th>\n",
       "      <td>Other</td>\n",
       "      <td>Monocyte</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Other</td>\n",
       "      <td>8042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1470</th>\n",
       "      <td>Other</td>\n",
       "      <td>Neutrophile</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Other</td>\n",
       "      <td>7806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_1471</th>\n",
       "      <td>Other</td>\n",
       "      <td>Neutrophile</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Other</td>\n",
       "      <td>2297</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>452 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Biosample type                  Biosample name Tissue type  \\\n",
       "Sample ID                                                                \n",
       "Sample_0025   Primary Cell  CD56Bright natural killer cell       Blood   \n",
       "Sample_0026   Primary Cell  CD56Bright natural killer cell       Blood   \n",
       "Sample_0027   Primary Cell  CD56Bright natural killer cell       Blood   \n",
       "Sample_0028   Primary Cell  CD56Bright natural killer cell       Blood   \n",
       "Sample_0029   Primary Cell     CD56Dim natural killer cell       Blood   \n",
       "...                    ...                             ...         ...   \n",
       "Sample_1467          Other                      Macrophage       Blood   \n",
       "Sample_1468          Other                        Monocyte       Blood   \n",
       "Sample_1469          Other                        Monocyte       Blood   \n",
       "Sample_1470          Other                     Neutrophile       Blood   \n",
       "Sample_1471          Other                     Neutrophile       Blood   \n",
       "\n",
       "            Cancer type  Region number  \n",
       "Sample ID                               \n",
       "Sample_0025      Normal          48025  \n",
       "Sample_0026      Normal          50447  \n",
       "Sample_0027      Normal          47349  \n",
       "Sample_0028      Normal          46698  \n",
       "Sample_0029      Normal          40745  \n",
       "...                 ...            ...  \n",
       "Sample_1467       Other          10568  \n",
       "Sample_1468       Other           6102  \n",
       "Sample_1469       Other           8042  \n",
       "Sample_1470       Other           7806  \n",
       "Sample_1471       Other           2297  \n",
       "\n",
       "[452 rows x 5 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples_frame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "2c917ca3",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_of_megaseqs.to_csv('megasequence-to-cancer.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "848dd340",
   "metadata": {},
   "outputs": [],
   "source": [
    "compression_opts = dict(method='zip', archive_name='megasequence-to-cancer.csv')\n",
    "df_of_megaseqs.to_csv('megasequence-to-cancer.zip', compression=compression_opts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "173cbd7b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
